types/stylelint/index.d.ts

```diff
@@ -252,6 +253,26 @@ declare namespace stylelint {
 		fixable?: boolean;
 	};
 
+	type FixerArguments = { [x: string]: any };
+	type Range = {
+		start?: {
+			line: number;
+			column?: number /** inclusive */;
+		};
+		end: {
+			line: number;
+			column?: number /** exclusive */;
+		};
+	};
+
+	export type Fixer = (args?: FixerArguments) => Range;
+	export type FixerData = {
+		source: { start: Required<Exclude<Range['start'], undefined>>; end: Required<Range['end']> };
+		callback: Fixer;
+		args?: FixerArguments;
```

Comment:
See the alternative API in <https://github.com/stylelint/stylelint/pull/7706#discussion_r1617499836> that I suggested. If we eventually expose a new API, the API should be usable.

Reply:
If we fix we don't report and vice versa.
i.e. you only report if you don't fix

So I don't understand why you would give the responsibility of the _fixer callback addition_ to `report`.
This seems illogical to me.

<hr />

Do you want me to 

> put `fixSimple` and `fixComplex` in the scope of l91–136
> remove `args` from `FixerData` type

?

Reply:
Let's go back to your original question:

> Is `args` necessary? Cannot we add a callback as a closure?

Do you want me to remove `args`?
It was intended as a nicety so that you don't have to move your fixers in the scope.
I am fine with it.

Reply:
> Is `args` necessary? Cannot we add a callback as a closure?

I went ahead and removed `args`.